File: app/src/main/java/io/ipoli/assistant/modules/AppModule.java
package io.ipoli.assistant.modules;

import android.content.Context;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.ipoli.assistant.app.App;

/**
 * Created by Venelin Valkov <venelin@curiousily.com>
 * on 1/7/16.
 */
@Module
public class AppModule {

    private final App app;

    public AppModule(App app) {
        this.app = app;
    }

    @Provides
    @Singleton
    public Context provideContext() {
        return app;
    }
}

File: app/src/main/java/io/ipoli/android/di/AppModule.kt
package io.ipoli.android.di

import android.content.Context
import dagger.Module
import dagger.Provides
import io.ipoli.android.player.PlayerRepository
import javax.inject.Singleton

/**
 * Created by vini on 8/1/17.
 */
@Module
class AppModule(val context: Context) {

    @Provides
    @Singleton
    fun provideContext(): Context = context
}